airflow/providers/google/cloud/hooks/cloud_build.py

```diff
@@ -158,13 +159,17 @@ def create_build_without_waiting_for_result(
         :param timeout: Optional, the amount of time, in seconds, to wait for the request to complete.
             Note that if `retry` is specified, the timeout applies to each individual attempt.
         :param metadata: Optional, additional metadata that is provided to the method.
+        :param location: Optional, The location of the project.
+            If set to None or missing, global is used.
```

Comment:
What does “global” mean here? It may be worthwhile to clarify in the docstring. (Or maybe I’m just missing context here, I’m not very familiar with the Google Cloud Platform.

Reply:
@uranusjr 
CloudBuild locations have global (non-regional) apart from general regions.
The current operator can only create builds on global which is the default.
This PR is to be able to change that.

https://cloud.google.com/sdk/gcloud/reference/builds/submit
```
--region=REGION
The region of the Cloud Build Service to use.
Must be set to a supported region name (e.g. us-central1).
If unset, builds/region, which is the default region to use when working with Cloud Build resources, is used.
If builds/region is unset, region is set to global.
```

However, when I actually specified the location with python client and tried to [create_build](https://cloud.google.com/python/docs/reference/cloudbuild/latest/google.cloud.devtools.cloudbuild_v1.services.cloud_build.CloudBuildClient#google_cloud_devtools_cloudbuild_v1_services_cloud_build_CloudBuildClient_create_build), an error occurred and I could not change it.
```
google.api_core.exceptions.InvalidArgument: 400 generic::invalid_argument: invalid value for 'parent': location "us-central1" is unsupported, service location is "global"
```

I also tried [list_builds 'parent'](https://github.com/apache/airflow/blob/f0bd45389c7799884a26d2b9c79b0498683d3e03/airflow/providers/google/cloud/hooks/cloud_build.py#L455) used in CloudBuildListBuildsOperator , but the global one was returned and didn't seem to work.

I am contacting GCP support about this issue.
Please wait for a while until the reply comes.

Reply:
I got a response from GCP support.
As a result, it turns out that the api_endpoint must be set on the client when using a private pool or a non-global default pool.
https://github.com/googleapis/python-cloudbuild/blob/9cf4f4269b55213c1f84206923d3cdc0308ecf57/samples/snippets/quickstart.py#L35-L43

Sample code
```
from google.cloud.devtools import cloudbuild_v1
client = cloudbuild_v1.CloudBuildClient(client_options={"api_endpoint":f"{region}-cloudbuild.googleapis.com:443"})
parent = f"projects/{project}/locations/{region}"
request = cloudbuild_v1.CreateBuildRequest(build={"steps":[{"name":"ubuntu", "args":["echo", "hello"]}]}, parent=parent)
operation = client.create_build(request)
print(operation.result())
```

Similarly, I've found that some operators may not work with private pools or a non-global default pool, but I'll fix that in another PR.
```
CloudBuildCancelBuildOperator
CloudBuildGetBuildOperator
CloudBuildListBuildsOperator
CloudBuildRetryBuildOperator
```

Reply:
@uranusjr @pankajastro
I fixed it, so please re-review
https://github.com/apache/airflow/pull/29689/commits/8a0b53943a392dd563c3896b3e23dc945d1c1ec7